src/quick-lint-js/parse.h

```diff
@@ -633,6 +663,9 @@ class parser {
 
     case token_type::end_of_file:
     case token_type::right_curly:
+      if (this->depth_ > 0) {
+        this->depth_--;
+      }
```

Comment:
What is this for? What is this supposed to match? This code looks misplaced, but maybe I'm not seeing something.

Reply:
Yup it was misplaced.